lr-cli/cortx_setup/commands/node/prepare/storage.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from cortx_setup.commands.command import Command
+from cortx_setup.commands.storage import Commons
+from cortx_setup.config import CONFSTORE_CLUSTER_FILE
+from cortx_setup.commands.common_utils import get_pillar_data
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id
+from provisioner.salt import function_run
+
+
+class NodePrepareStorage(Command):
+    """Cortx Setup API for Preparing Storage in field"""
+    _args = {
+        'user': {
+            'type': str,
+            'default': None,
+            'optional': True
+        },
+        'password': {
+            'type': str,
+            'default': None,
+            'optional': True
+        }
+    }
+
+    def validate_user_password(
```

Comment:
I guess the method name is `validate_credentials`

Reply:
done